roles/dnsmasq/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+---
+- name: install dnsmasq and bind-utils
+  sudo: yes
+  yum:
+    name: "{{ item }}"
+    state: latest
+  with_items:
+    - dnsmasq
+    - bind-utils
+  when: inventory_hostname in groups[master_group_name]
+  tags:
+    - dnsmasq
+
+- name: install networkmanager
```

Comment:
assuming the question is installing on nodes works, then this could be appended to the previous then.